### PR TITLE
SQL: ODBC: document extra connection string parameters

### PR DESCRIPTION
**Pull request by bpintea**
This PR documents the connection string parameters of the ODBC driver that are currently not configurable through the GUI.

It also adds a note about the SmartScreen possible warning on driver MSI installation and one about the TLS certificate file not being supported as bundled or password-protected.

### CODE REVIEW

File: docs/reference/sql/endpoints/odbc.asciidoc [hunk 1]
```diff
@@ -9,9 +9,12 @@
...
 === Overview
 
-{odbc} is a feature-rich 3.80 ODBC driver for {es}.
-It is a core level driver, exposing all of the functionality accessible through the {es}'s SQL ODBC API, converting ODBC calls into
-{es-sql}.
+{odbc} is a 3.80 compliant ODBC driver for {es}.
+It is a core level driver, exposing all of the functionality accessible through
+the {es}'s SQL ODBC API, converting ODBC calls into {es-sql}.
++
+In order to be able to use of the driver, the server must have the {es-sql}
```
Comment (costin) [inline docs/reference/sql/endpoints/odbc.asciidoc]: `must have {es-sql} installed and running with the valid license`.

File: docs/reference/sql/endpoints/odbc/configuration.asciidoc [hunk 2]
```diff
@@ -260,7 +262,110 @@ image:images/sql/odbc/env_var_log.png[]
...
+The following is a list of additional parameters that can be configured for a
+particular connection, in case the default behavior of the driver is not
+suitable. This can be done within the client application, in a manner
+particular to that application, generally in a free text input box (sometimes
+named "Connection string", "String extras", or similar).  The format of the
+string is `Attribute1=Value1`. Multiple attributes can be specified, separated
+by a semicolon `Attribute1=Value1;Attribute2=Value2;`. The attribute names are
+given below.
+
+===== Timeout
```
Comment (costin) [inline docs/reference/sql/endpoints/odbc/configuration.asciidoc]: I suggest using either a table or like in the JDBC chapter a list by using `entry::` syntax: https://github.com/elastic/elasticsearch/blob/47e71b988ee550409ef1863354061f17d298050d/docs/reference/sql/endpoints/jdbc.asciidoc#essential
Also do note the use of links `[[link-name]]` which are useful for linking parts of the docs directly

**Review by costin — APPROVED** [review]
LGTM

File: docs/reference/sql/endpoints/odbc.asciidoc (hunk 1 above)
Comment (bpintea) [reply docs/reference/sql/endpoints/odbc.asciidoc]: Thanks, fixed.

File: docs/reference/sql/endpoints/odbc/configuration.asciidoc (hunk 2 above)
Comment (bpintea) [reply docs/reference/sql/endpoints/odbc/configuration.asciidoc]: Great point, used entries.
The link seems to only works for a section (not entries), so I've added one for the entire section.

File: docs/reference/sql/endpoints/odbc/configuration.asciidoc [hunk 3]
```diff
@@ -260,7 +262,110 @@ image:images/sql/odbc/env_var_log.png[]
...
+
+* `major`: the versions must have the same major number.
+
+Default: `strict`.
+
+===== MultiFieldLenient
+
+This boolean parameter controls the behavior of the server in case a
+multi-value field is queried. In case this is set and the server encounters
+such a field, it will pick the first value in the set and return it as the
```
Comment (astefan) [inline docs/reference/sql/endpoints/odbc/configuration.asciidoc]: I would add the same wording used for the REST API (I'm working on adding a note for JDBC, as well) about what "first value" means:
> (without any guarantees of what that will be - typically the first in natural ascending order)

From https://www.elastic.co/guide/en/elasticsearch/reference/master/sql-rest.html.

**Review by astefan — APPROVED** [review]
LGTM. Left some comments.

File: docs/reference/sql/endpoints/odbc.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (bpintea) [reply docs/reference/sql/endpoints/odbc.asciidoc]: rephrased, thanks.

File: docs/reference/sql/endpoints/odbc.asciidoc [hunk 4]
```diff
@@ -9,9 +9,12 @@
 [float]
 === Overview
 
-{odbc} is a feature-rich 3.80 ODBC driver for {es}.
-It is a core level driver, exposing all of the functionality accessible through the {es}'s SQL ODBC API, converting ODBC calls into
-{es-sql}.
+{odbc} is a 3.80 compliant ODBC driver for {es}.
+It is a core level driver, exposing all of the functionality accessible through
+the {es}'s SQL ODBC API, converting ODBC calls into {es-sql}.
```
[the remark this answers is not in the document]
Comment (bpintea) [reply docs/reference/sql/endpoints/odbc.asciidoc]: The `ODBC` was actually superfluous => `Elasticsearch's SQL API`

File: docs/reference/sql/endpoints/odbc/configuration.asciidoc [hunk 5]
```diff
@@ -260,7 +262,110 @@ image:images/sql/odbc/env_var_log.png[]
...
+* `default`: the default notation (ex.: 12.3);
+
+* `auto`: the driver will chose one of the above depending on the value to be
+printed.
+
+Note that the number of decimals is dependent on the precision (or ODBC scale)
+of the value being printed and varies with the different floating point types
+supported by {es-sql}.
+This setting is not effective when the application fetches from the driver the
+values as numbers itself and then does the conversion subsequently.
```
[the remark this answers is not in the document]
Comment (bpintea) [reply docs/reference/sql/endpoints/odbc/configuration.asciidoc]: moved the `itself` to the end.

File: docs/reference/sql/endpoints/odbc/configuration.asciidoc (hunk 3 above)
Comment (bpintea) [reply docs/reference/sql/endpoints/odbc/configuration.asciidoc]: Good point, thanks. I've also added the cross ref to paragraph with the parameters description.

**Pull request merged**